### PR TITLE
[backend/frontend] Danger zone: Remove FF (#8284)

### PR DESCRIPTION
**Pull request by marieflorescontact**
### Proposed changes

* Removes Feature flag
* documentation PR: https://github.com/OpenCTI-Platform/docs/pull/217
* Fixes 2 css issues in RuleList

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/8284

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case (coverage and e2e)
- [x] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/settings/RulesList.jsx [hunk 1]
```diff
@@ -549,7 +549,7 @@ const RulesListComponent = ({ relay, data, keyword }) => {
                     <Paper
                       variant="outlined"
                       classes={{ root: classes.paper }}
-                      style={{ marginTop: 25 }}
+                      style={{ marginTop: 23 }}
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/private/components/settings/RulesList.jsx]: I don't think it's the right solution for the display issue, even though it seem to work. I think the rule title should be moved outside of the grid, and we should remove completly this marginTop here.

Comment (SouadHadjiat) [reply opencti-platform/opencti-front/src/private/components/settings/RulesList.jsx]: also it would be really nice to add a overflowX: 'auto' here so that it finally displays the whole rule definition for small screens.

Comment (marieflorescontact) [reply opencti-platform/opencti-front/src/private/components/settings/RulesList.jsx]: > I don't think it's the right solution for the display issue, even though it seem to work. I think the rule title should be moved outside of the grid, and we should remove completly this marginTop here.

I don't know how to make the change because the rule name is now part of the `DangerZoneBlock` component 😕

File: opencti-platform/opencti-front/src/private/components/settings/groups/Group.tsx [hunk 2]
```diff
@@ -467,7 +467,7 @@ const Group = ({ groupData }: { groupData: Group_group$key }) => {
       </Grid>
       <GroupEdition
         groupId={group.id}
-        disabled={!isAllowed}
+        disabled={!isAllowed && isSensitive}
```
Comment (labo-flg) [inline opencti-platform/opencti-front/src/private/components/settings/groups/Group.tsx:470]: why this change ?
I would naively think that isAllowed is always true if isSensitive is false (a non-sensitive zone is always granted).

**Review by Archidoit — DISMISSED** [review]

**Comment by Archidoit** [comment]
The Danger Zone button doesn't seem well aligned with the text (it is a bit lower)

image

File: opencti-platform/opencti-front/src/private/components/settings/groups/Group.tsx (hunk 2 above)
Comment (marieflorescontact) [reply opencti-platform/opencti-front/src/private/components/settings/groups/Group.tsx:470]: because the edit button was disabled for non sensitive Group/Role if user was not allowed (not allowed to modified sensitive Group/Role )

**Comment by marieflorescontact** [comment]
> The Danger Zone button doesn't seem well aligned with the text (it is a bit lower)
> 
> image

well in fact this is an illusion:
Capture d'écran 2024-11-07 101153

**Comment by Archidoit** [comment]
> > The Danger Zone button doesn't seem well aligned with the text (it is a bit lower)
> > image
> 
> well in fact this is an illusion: Capture d'écran 2024-11-07 101153

yes the text in the button is aligned with the text, but the button itself is not aligned... Only design question I don't know if we let that :)

**Comment by marieflorescontact** [comment]
> yes the text in the button is aligned with the text, but the button itself is not aligned... Only design question I don't know if we let that :)

 I don't see what is not aligned, could you explain? 💐

**Review by Archidoit — APPROVED** [review]
